### DISCUSSION TITLE
Static IP for clients

### DISCUSSION THREAD

**Discussion by ant7511** [angristan/openvpn-install#851] [Help]
Hello, 

I wanted to give a static ip to each clients, to set the permission on my home network. 
I've read many articles about it, but i used the openvpn install script, and it creates the certs in different way as in the articles, so i'm unable to add static ip to the clients.  Is there any tutorial how to do that ?

**Comment by impulsado** [comment]
Between this code
```bash
[...]
# Generates the custom client.ovpn
	cp /etc/openvpn/client-template.txt "$homeDir/$CLIENT.ovpn"
	{
		echo "<ca>"
		cat "/etc/openvpn/easy-rsa/pki/ca.crt"
		echo "</ca>"
[...]
```
Add this commands, where $CLIENT_IP and $vpn_mask must be defined manually.
```bash
[...]
# Generates the custom client.ovpn
	echo "ifconfig-push ${CLIENT_IP} ${vpn_mask}" > /etc/openvpn/ccd/${CLIENT}  # ASSIGN USER STATIC IPv4
	cp /etc/openvpn/client-template.txt "$homeDir/$CLIENT.ovpn"
	{
		echo "<ca>"
		cat "/etc/openvpn/easy-rsa/pki/ca.crt"
		echo "</ca>"
[...]
```

I know that could be solved better, but this is the key command.

Hope it helps :)

**Comment by superspiff83** [comment]
Hello,
I know that the topic is a bit old but anyway, it is a question I also had and as I am not good enough to apply Impulsado method, I figured out a solution which is quite simple. But it will work only if you don't have thousands of static IP to set.
After having used openvpn-install script to add user you can create a file for each user who should have a static IP in the /etc/openvpn/ccd folder.
`nano /etc/openvpn/ccd/USERNAME`
Then in the file just write this line
`ifconfig-push STATIC_IP SUBNETMASK`
Where STATIC_IP is the IP you want to set for the User and SUBNETMASK is the OpenVPN server's subnet mask, you can retrieve it from the /etc/openvpn/server.conf file
It worked for me. 
I hope it can help other people.

**Reply by yudiz-Manushi** [reply]
@superspiff83 
I'm using latest version of openvpn.
I have configured a same sol that you provided.
I have one client file and 3 users(user1,user2,user3).
I have created ccd dir and put user1 file in it and put ifconfig details in it.

But still not able to do it like user assign diff IP other than I have defined in user1 file.

Let me know if you have any idea about it.

**Reply by superspiff83** [reply]
Hi @yudiz-Manushi,
Unfortunately I am not very good at setting OpenVPN servers. So I am not sure to be helpful here.
Did you try to restart the OpenVPN server after having created the user files ?

**Reply by yudiz-Manushi** [reply]
Hi @superspiff83 
Yes I have restarted openvpn server.
